rules/fusesoc.bzl

```diff
@@ -36,15 +36,20 @@ def _fusesoc_build_impl(ctx):
         verilator_options = ctx.attr.verilator_options[BuildSettingInfo].value
         flags.append("--verilator_options={}".format(" ".join(verilator_options)))
 
+    cores_roots = {}
+    for c in ctx.files.cores:
+        if c.dirname not in cores_roots:
+            cores_roots[c.dirname] = True
+
     # Note: the `fileset_top` flag used above is specific to the OpenTitan
     # project to select the correct RTL fileset.
     ctx.actions.run(
         mnemonic = "FuseSoC",
         outputs = outputs,
         inputs = ctx.files.srcs + ctx.files.cores,
         arguments = [
```

Comment:
Not to pile on too much to this PR, but would you mind refactoring the argument passing to use Bazel's [Args object](https://bazel.build/rules/lib/Args)?

Reply:
I'll take a look

Reply:
I've done this in #15554 Should I merge this one and rebase it or would you prefer to wait for it all at once?

Reply:
I'm fine with merging them separately